src/main/java/pl/javastart/task/Person.java

```diff
@@ -0,0 +1,76 @@
+package pl.javastart.task;
+
+public class Person {
+    private String firstName;
+    private String lastName;
+    private int age;
+    private String pesel;
+
+    public Person(String firstName, String lastName, int age, String pesel) {
+        setFirstName(firstName);
+        setLastName(lastName);
+        setAge(age);
+        this.pesel = pesel;
+    }
+
+    private static void checkAge(int age) {
+        if (age < 1) {
+            throw new IncorrectAgeException("Nieprawidłowy wiek");
+        }
+    }
+
+    private static void validateLastName(String lastName, String message) {
+        if (lastName == null || lastName.length() < 2) {
+            throw new NameUndefinedException(message);
+        }
+    }
+
+    private static void validateFirstName(String firstName, String message) {
+        validateLastName(firstName, message);
+    }
```

Comment:
Jeżeli chcesz się pozbyć duplikacji, to ok, ale nie w ten sposób. Z Twojej metody validateFirstName wynika, że sprawdzasz w niej z jakiegoś powodu nazwisko. Oczywiście rozumiem o co chodzi, ale jest to mało czytelne.
Lepiej byłoby wydzielić powtarzające się reguły do osobnych metod i mieć np. coś takiego:
```
    private static void validateLastName(String lastName, String message) {
         if (isNotNullAndMinLength(2)) {
             throw new NameUndefinedException(message);
         }
     }
...
```
W ten sposób wprowadzasz też większą elastyczność i gdy np. zmienią się reguły, tzn. np. dla imienia będą wymagane minimum 3 znaki, a dla nazwiska 2, to zmiana będzie łatwiejsza.
Te metody też nie muszą być statyczne, szczególnie że są prywatne i nie będą używane poza tą klasą.

Reply:
Poprawię. Akurat z tym to musiałem sam coś poprzeklikać, pewnie coś wyszło podczas kopiowania. Sam jestem zdziwiony, że coś takiego tam się znalazło.